kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -0,0 +1,368 @@
+package kyo
+
+export Async.Fiber
+export Async.Promise
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.LockSupport
+import kyo.Maybe.Empty
+import kyo.Result.Panic
+import kyo.Tag
+import kyo.internal.FiberPlatformSpecific
+import kyo.kernel.*
+import kyo.scheduler.*
+import scala.annotation.implicitNotFound
+import scala.annotation.tailrec
+import scala.annotation.targetName
+import scala.collection.immutable.ArraySeq
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.util.NotGiven
+import scala.util.control.NonFatal
+import scala.util.control.NoStackTrace
+
+opaque type Async <: (IO & Async.Join) = Async.Join & IO
+
+object Async:
+
+    sealed trait Join extends Effect[IOPromise[?, *], Result[Nothing, *]]
+
+    inline def run[E, A: Flat, Ctx](inline v: => A < (Abort[E] & Async & Ctx))(
```

Comment:
`Fibers` used to have `run` and `init` with a similar role returning `Fiber` instances. I've kept only `run` in this new design.

Reply:
I've followed the naming in [`Boundary`](https://github.com/getkyo/kyo/blob/719d67a411562aa540dc7f21d9e44b03d73e161a/kyo-prelude/shared/src/main/scala/kyo2/kernel/Boundary.scala#L8). It makes it easier for me to reason about the code because `S` typically represents an unconstrained set of pending effects. Since the set in this case is limited to `ContextEffect`s, naming it `Ctx` better indicates what the set contains. For example, someone looking at `IOTask` and seeing an `S` type param might think that it can contain arbitrary effects. Using `Ctx` there as well I think makes the code clearer.